### PR TITLE
docs: Add table of supported platforms

### PR DESCRIPTION
**Pull request by rleigh-codelibre**
Initial documentation for the C++ platform support.

### CODE REVIEW

File: docs/sphinx/platforms.rst [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
++---------------+--------------+--------------+--------------+-------------+
+| Ubuntu        | 16.04        | 14.04        | Supported    | Supported   |
++---------------+--------------+--------------+--------------+-------------+
+| Windows       | VS2015       | VS2013       | Unsupported§ | Supported   |
++---------------+--------------+--------------+--------------+-------------+
+
+\*
+  CentOS does not provide all the prequired libraries; CentOS 6
+  provides a broken version of Boost and an obsolete libtiff, so using
+  the Super-Build or building new versions of the needed libraries is
```
Comment (mtbc) [inline docs/sphinx/platforms.rst]: we usually don't capitalize "super-build", may want to ensure the docs are consistent in this (also below)

File: docs/sphinx/platforms.rst [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+§
+  Windows does not have a package manager or provide any third-party
+  libraries for a given Visual Studio version; either use the
+  Super-Build or build the needed libraries yourself.
+
+"Platform" builds are builds of the OME C++ components with
+third-party library and tool dependencies provided by the platform,
+i.e. the operating system's package manager.  Systems without a
+package manager, or with missing or outdated packages, can not support
```
Comment (mtbc) [inline docs/sphinx/platforms.rst]: cannot

File: docs/sphinx/platforms.rst [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+status for these is as follows:
+
++---------------+--------------+--------------+--------------+-------------+
+|               |           Version           |         Build type         |
++---------------+--------------+--------------+--------------+-------------+
+| System        | Recommended  | Minimum      | Platform     | Super-Build |
++===============+==============+==============+==============+=============+
+| CentOS        | 7            | 6            | Unsupported* | Supported   |
++---------------+--------------+--------------+--------------+-------------+
+| Debian        | 8            | 7            | Unsupported  | Unsupported |
```
Comment (sbesson) [inline docs/sphinx/platforms.rst]: As both are unsupported, I would suggest we simply drop Debian from this table.

File: docs/sphinx/platforms.rst [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
++---------------+--------------+--------------+--------------+-------------+
+| FreeBSD       | 11           | 10           | Supported†   | Supported   |
```
Comment (sbesson) [inline docs/sphinx/platforms.rst:18]: Do we have platform tests for FreeBSD? I thought it was superbuild only.

File: docs/sphinx/platforms.rst [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  CentOS does not provide all the prequired libraries; CentOS 6
+  provides a broken version of Boost and an obsolete libtiff, so using
+  the Super-Build or building new versions of the needed libraries is
+  a requirement.
+
+†
+  With 11 only; 10 may work with ports built with a newer toolchain.
+
+‡
+  Supported via homebrew.
```
Comment (sbesson) [inline docs/sphinx/platforms.rst]: Homebrew. Note support should be upcoming with OME Files C++ 0.2.2

**Review by sbesson — CHANGES_REQUESTED** [review]
Generally renders well on https://www.openmicroscopy.org/site/support/ome-files-cpp-staging/ome-cmake-superbuild/manual/html/platforms.html and certainly helps documenting the decisions made in https://trello.com/c/J8affYoo/11-platform-support-and-build-rationalisation.

A couple of inline comments that require to be addressed/discussed. Otherwise this is mostly good to merge.

File: docs/sphinx/platforms.rst (hunk 4 above)
Comment (rleigh-codelibre) [reply docs/sphinx/platforms.rst:18]: No, this one verified by hand (I've started a test build to confirm, and we can add it back to the matrix if it's OK).  This is the same for Ubuntu 16.04; the building here is local in the absence of CI support.

File: docs/sphinx/platforms.rst (hunk 5 above)
Comment (rleigh-codelibre) [reply docs/sphinx/platforms.rst]: Note this is unrelated to our providing homebrew packages; this is only about the dependencies.

**Review by sbesson — APPROVED** [review]

File: docs/sphinx/platforms.rst (hunk 3 above)
Comment (rleigh-codelibre) [reply docs/sphinx/platforms.rst]: Done

File: docs/sphinx/platforms.rst (hunk 4 above)
Comment (rleigh-codelibre) [reply docs/sphinx/platforms.rst:18]: FYI: https://ci.openmicroscopy.org/job/OME-FILES-CPP-DEV-merge/329/

**Pull request merged**